### ISSUE TITLE
[feature request] Text file preview

### ISSUE THREAD

**Issue by Avimitin** [obaland/vfiler.vim#2]
Is it possible to preview text file in a new vertical split window? It will be convenient to select file.

My expect output is like this:

![image](https://user-images.githubusercontent.com/30021675/148677927-d220fd23-8bd9-47c3-9b2e-d444cbb32529.png)

Or is it possible to draw two list in same window like the telescope?

![image](https://user-images.githubusercontent.com/30021675/148677966-bcc82637-0113-4b0a-b9c5-4ab22d4ce79b.png)

**Comment by obaland** [comment]
Thank for your request.
Sorry, currently there is no preview function.

I will serious consider your request.
I would like to add a "**preview**" action.

Please give me some time.
If you have any progress, I'll let you know here.

**Comment by obaland** [comment]
In https://github.com/obaland/vfiler.vim/commit/260c66653e8071988be78ed941ba5ecafc4f991a, Implemented the preview action.

| Key | Action | Description |
| --- | --- | --- |
| p | toggle_preview | Toggle the preview window for the item in the current cursor |
| C-P | toggle_auto_preview | Toggle the automatic preview window |

Please try using it.
Welcome your feedback.

**Comment by Avimitin** [comment]
That's so cool! But I got some error while moving cursor around:

![image](https://user-images.githubusercontent.com/30021675/150481572-979d4f47-1a80-4c3c-9cca-27ed101b7443.png)

And then the windows got polluted:

![image](https://user-images.githubusercontent.com/30021675/150481646-0588ed14-ab08-448c-8dab-d0f57c8f16bb.png)

Error occurs randomly, so it is hard to figure out which file cause this issue.

**Comment by Avimitin** [comment]
Besides, the neovim crash with SIGABRT when I moving around with preview toggle by key `p`. 

![image](https://user-images.githubusercontent.com/30021675/150482618-539c3253-34f6-4ab3-85f8-0995b87a66f3.png)

**Comment by obaland** [comment]
I want to start by apologising.
I will try to find out if it can be reproduced here as well.
Can you tell me which version of neovim you are using?

**Comment by Avimitin** [comment]
I am using `NVIM v0.7.0-dev+907-g96aa7dd097`. And the file I am browsing is the my [neovim configuration](https://github.com/Avimitin/nvim).

**Comment by obaland** [comment]
Fixed and improved it in 38dd489104b19c54c028d3c8923b2f921b0ccdc2, but the "toggle_auto_preview" action is still not stable. I suspect a conflict with other "CursorMoved" or "CursorHold" events. In conclusion, the "toggle_auto_preview" action is temporarily abolished, as the radical fix will still take some time.

I'm sorry, but please use the "toggle_preview" action for a while. We'll let you know when the "toggle_auto_preview" action comes back.

**Comment by Avimitin** [comment]
You remind me. I am using the https://github.com/antoinemadec/FixCursorHold.nvim plugin. Maybe this is the reason of this issue. You can take a look on this performance issue: https://github.com/neovim/neovim/issues/12587#issuecomment-1005279969

**Comment by obaland** [comment]
Thank you for providing the information.
4227d8fabec27097cf987a55c4145668f0bdad40 has corrected the buffer reading. I think your report has improved.
Please try it.

**Comment by Avimitin** [comment]
The single file preview works well. And auto preview works well too.

But there is another bug. When I am using auto preview and use the `q` key to quit, an error prompts up:

```text
E5108: Error executing lua ...im/site/pack/packer/opt/vfiler.vim/lua/vfiler/buffer.lua:31: Vim(bwipeout):E444: Cannot close last window
stack traceback:
        [C]: in function 'command'
        ...im/site/pack/packer/opt/vfiler.vim/lua/vfiler/buffer.lua:31: in function 'delete'
        ...nvim/site/pack/packer/opt/vfiler.vim/lua/vfiler/view.lua:110: in function 'delete'
```

All window will quit successfully, but the error might need to be handled.

---

Anyway, you have did a great job. My sincere thanks for your efforts。

**Comment by obaland** [comment]
Thank you for your report.
Sorry for taking up so much of your time.
I'll fix it and contact you.

**Comment by obaland** [comment]
I fixed it with 15a68331e308ba778eafc9dfdf8d7c544ee9e24c.
Please try it.

**Comment by Avimitin** [comment]
Thanks! It works perfectly now!
